src/AsyncElegantOTA.cpp

```diff
@@ -111,10 +111,19 @@ void AsyncElegantOtaClass::loop() {
 }
 
 void AsyncElegantOtaClass::restart() {
-    yield();
-    delay(1000);
-    yield();
-    ESP.restart();
+    #if defined CONFIG_IDF_TARGET_ESP32S2 || defined CONFIG_IDF_TARGET_ESP32S2
+        yield();
+        delay(1000);
+        yield();
+        esp_task_wdt_init(1,true);
+        esp_task_wdt_add(NULL);
+        while(true);
+    #else
+        yield();
+        delay(1000);
+        yield();
+        ESP.restart();
+    #endif
```

Comment:
Since the basic example works this seems likely to be a bug elsewhere in your code/the code you rely on, rather than something that this library should deal with.

If this workaround is actually necessary, you should structure the code so it's clear what is actually different between the branches:
```suggestion
    yield();
    delay(1000);
    yield();
    #if defined(CONFIG_IDF_TARGET_ESP32S2)
        // Normal restarts don't work due to X. Abuse the watchdog timer to perform a hard reset.
        esp_task_wdt_init(1, true);
        esp_task_wdt_add(NULL);
        while(true);
    #else
        ESP.restart();
    #endif
```